onset_maproom/tests/test_calc.py

```diff
@@ -4,10 +4,463 @@
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.2552221 ,  0.12695578,  0.10042991,  0.00686576,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.06907884,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.00165216,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.21126945,  5.749138  ,
+        5.4374275 ,  3.3610737 ,  0.00103077,  0.08556359,  0.07880804,
+        2.277596  ,  0.11416379,  0.        ,  0.        ,  0.        ,
+        0.        ,  4.0256405 ,  1.322553  ,  0.        ,  0.2692001 ,
+        0.        ,  0.        ,  1.7372526 ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.45122755,  0.05431353,  0.33383146,  0.6377049 ,
+        6.4067054
+    ]
+    # fmt: on
+    precip = xr.DataArray(values, dims=["T"], coords={"T": t}).rename("precip")
+    return precip
+
+
+def test_daily_tobegroupedby_season_cuts_on_days():
+
+    precip = set_multi_year_data_sample()
+    dts = calc.daily_tobegroupedby_season(precip, 29, 11, 29, 2)
+
+    assert dts["T"].size == 461
+
+
+def test_daily_tobegroupedby_season_creates_groups():
+
+    precip = set_multi_year_data_sample()
+    dts = calc.daily_tobegroupedby_season(precip, 29, 11, 29, 2)
+
+    assert dts["group"].size == 5
+
+
+def test_daily_tobegroupedby_season_picks_right_end_dates():
+
+    precip = set_multi_year_data_sample()
+    dts = calc.daily_tobegroupedby_season(precip, 29, 11, 29, 2)
+    assert (
+        dts.seasons_ends
+        == pd.to_datetime(
+            xr.DataArray(
+                [
+                    "2001-02-28T00:00:00.000000000",
+                    "2002-02-28T00:00:00.000000000",
+                    "2003-02-28T00:00:00.000000000",
+                    "2004-02-29T00:00:00.000000000",
+                    "2005-02-28T00:00:00.000000000",
+                ],
+                dims=["group"],
+                coords={"group": dts.seasons_ends["group"]},
+            )
+        )
+    ).all()
+
+
+def test_seasonal_onset_date_keeps_returning_same_outputs():
```

Comment:
In response to https://github.com/iridl/x24-vector/pull/20#discussion_r729641337 :
> I was not sure what really to test here. 

You are asking the right questions. There are no definitive answers. I agree that this test is fine the way it is, as a canary to alert us to something that changes unexpectedly.

I think this test is a good one to run on real data. The other tests set up a particular scenario and then test for desired behavior in that scenario; those would be better converted to use synthetic data, to make it easier for the reader to undertand the scenario. For this test we're not choosing particular conditions, we're just checking that nothing has changed, so (as you said) it's not clear what synthetic data we would choose.